RATapi/inputs.py

```diff
@@ -154,6 +154,23 @@ def make_problem(project: RATapi.Project) -> ProblemDefinition:
     hydrate_id = {"bulk in": 1, "bulk out": 2}
     prior_id = {"uniform": 1, "gaussian": 2, "jeffreys": 3}
 
+    # Ensure backgrounds and resolutions have a source defined
+    for contrast in project.contrasts:
+        background = project.backgrounds[contrast.background]
+        resolution = project.resolutions[contrast.resolution]
+        if background.source == "":
+            raise ValueError(
+                f"All backgrounds must have a source defined. For a {background.type} type background, "
+                f"the source must be defined in "
+                f'"{RATapi.project.values_defined_in[f"backgrounds.{background.type}.source"]}"'
+            )
+        if resolution.source == "" and resolution.type != TypeOptions.Data:
+            raise ValueError(
+                f"All constant and function resolutions must have a source defined. For a "
```

Comment:
There is no function resolution so maybe we should leave that out of the error message.

Reply:
Good catch, thanks for that!